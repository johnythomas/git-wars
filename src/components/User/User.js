import React from "react";
import PropTypes from "prop-types";
import { Image, Grid } from "semantic-ui-react";

const User = ({ user }) => (
  <Grid centered>
    <Grid.Row>
      <Image
        src={user.avatar_url}
        style={{ borderRadius: 200, width: 200, height: 200 }}
      />
    </Grid.Row>
    <Grid.Row>{`Name: ${user.name}`}</Grid.Row>
    <Grid.Row>{`Location: ${user.location}`}</Grid.Row>
    <Grid.Row>{`Followers: ${user.followers}`}</Grid.Row>
    <Grid.Row>{`Following: ${user.following}`}</Grid.Row>
  </Grid>
);

User.propTypes = {
  user: PropTypes.shape({
    name: PropTypes.string.isRequired,
    avatar_url: PropTypes.string,
    location: PropTypes.string.isRequired,
    followers: PropTypes.number.isRequired,
    following: PropTypes.number.isRequired
  }).isRequired
};

export default User;
