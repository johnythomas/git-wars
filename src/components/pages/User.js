import React from "react";
import { Image, Card, Icon } from "semantic-ui-react";

const User = () => (
  <Card>
    <Image src="https://avatars3.githubusercontent.com/u/6956481?s=460&v=4" />
    <Card.Content>
      <Card.Header>Johny</Card.Header>
      <Card.Description>User Details</Card.Description>
    </Card.Content>
    <Card.Content extra>
      <a>
        <Icon name="user" />
        Follow
      </a>
    </Card.Content>
  </Card>
);

export default User;
