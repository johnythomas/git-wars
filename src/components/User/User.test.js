import React from "react";
import { shallow } from "enzyme";
import User from "./User";

describe("User", () => {
  it("should render correctly", () => {
    const user = {
      name: "John Doe",
      avatar_url: "http://github.com/johnDoe",
      location: "US",
      followers: 100,
      following: 10
    };
    const wrapper = shallow(<User user={user} />);
    expect(wrapper).toMatchSnapshot();
  });
});
