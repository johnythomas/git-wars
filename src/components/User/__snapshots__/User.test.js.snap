// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <User
    user={
        Object {
            "avatar_url": "http://github.com/johnDoe",
            "followers": 100,
            "following": 10,
            "location": "US",
            "name": "John Doe",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "centered": true,
      "children": Array [
        <GridRow>
          <Image
                    as="img"
                    src="http://github.com/johnDoe"
                    style={
                              Object {
                                        "borderRadius": 200,
                                        "height": 200,
                                        "width": 200,
                                      }
                    }
                    ui={true}
          />
</GridRow>,
        <GridRow>
          Name: John Doe
</GridRow>,
        <GridRow>
          Location: US
</GridRow>,
        <GridRow>
          Followers: 100
</GridRow>,
        <GridRow>
          Following: 10
</GridRow>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Image
            as="img"
            src="http://github.com/johnDoe"
            style={
                        Object {
                                    "borderRadius": 200,
                                    "height": 200,
                                    "width": 200,
                                  }
            }
            ui={true}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "img",
            "src": "http://github.com/johnDoe",
            "style": Object {
              "borderRadius": 200,
              "height": 200,
              "width": 200,
            },
            "ui": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Name: John Doe",
        },
        "ref": null,
        "rendered": "Name: John Doe",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Location: US",
        },
        "ref": null,
        "rendered": "Location: US",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Followers: 100",
        },
        "ref": null,
        "rendered": "Followers: 100",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Following: 10",
        },
        "ref": null,
        "rendered": "Following: 10",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "centered": true,
        "children": Array [
          <GridRow>
            <Image
                        as="img"
                        src="http://github.com/johnDoe"
                        style={
                                    Object {
                                                "borderRadius": 200,
                                                "height": 200,
                                                "width": 200,
                                              }
                        }
                        ui={true}
            />
</GridRow>,
          <GridRow>
            Name: John Doe
</GridRow>,
          <GridRow>
            Location: US
</GridRow>,
          <GridRow>
            Followers: 100
</GridRow>,
          <GridRow>
            Following: 10
</GridRow>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Image
              as="img"
              src="http://github.com/johnDoe"
              style={
                            Object {
                                          "borderRadius": 200,
                                          "height": 200,
                                          "width": 200,
                                        }
              }
              ui={true}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "img",
              "src": "http://github.com/johnDoe",
              "style": Object {
                "borderRadius": 200,
                "height": 200,
                "width": 200,
              },
              "ui": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Name: John Doe",
          },
          "ref": null,
          "rendered": "Name: John Doe",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Location: US",
          },
          "ref": null,
          "rendered": "Location: US",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Followers: 100",
          },
          "ref": null,
          "rendered": "Followers: 100",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Following: 10",
          },
          "ref": null,
          "rendered": "Following: 10",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
